src/app/static/src/app/components/projects/ShareProject.vue

```diff
@@ -0,0 +1,122 @@
+            <h4 v-translate="'shareProject'"></h4>
+            <div v-if="!loading">
+                <div v-html="instructions" id="instructions"></div>
+                <div class="help-text text-muted small">E.g. someone@gmail.com, another@hotmail.com</div>
+                <input class="form-control"
+                       :class="{'is-invalid': showValidationFeedback}"
+                       v-model="emailsToShareWith"/>
+                <div class="invalid-feedback"
+                     v-translate="'emailMultiValidation'">
+                </div>
+            </div>
+            <div class="text-center" v-if="loading">
+                <loading-spinner size="sm"></loading-spinner>
+            </div>
+            <template v-slot:footer>
+                <button type="button"
+                        class="btn btn-red"
+                        @click="confirmShareProject"
+                        v-translate="'ok'">
+                </button>
+                <button type="button"
+                        class="btn btn-white"
+                        @click="cancelShareProject"
+                        v-translate="'cancel'">
+                </button>
+            </template>
+        </modal>
+    </div>
+</template>
+
+<script lang="ts">
+    import {Project} from "../../types";
+    import Vue from "vue";
+    import Modal from "../Modal.vue";
+    import LoadingSpinner from "../LoadingSpinner.vue";
+    import {mapStatePropByName, validateEmail} from "../../utils";
+    import i18next from "i18next";
+    import {Language} from "../../store/translations/locales";
+
+    interface Data {
+        emailsToShareWith: string
+        open: boolean
+        loading: boolean,
+        showValidationFeedback: boolean
+    }
+
+    interface Props {
+        project: Project
+    }
+
+    interface Computed {
+        currentLanguage: Language,
+        invalidEmail: boolean
+        instructions: string
+    }
+
+    interface Methods {
+        shareProject: (e: Event) => void
+        confirmShareProject: () => void
+        cancelShareProject: () => void
+    }
+
+    export default Vue.extend<Data, Methods, Computed, Props>({
+        props: {
+            project: {
+                type: Object
+            }
+        },
+        data() {
+            return {
+                emailsToShareWith: "",
+                open: false,
+                loading: false,
+                showValidationFeedback: false
+            }
+        },
+        methods: {
+            shareProject(e: Event) {
+                e.preventDefault();
+                this.open = true;
+            },
+            confirmShareProject() {
+                if (this.invalidEmail) {
+                    this.showValidationFeedback = true;
+                } else {
+                    this.showValidationFeedback = false;
+                    this.loading = true;
+                    const emails = this.emailsToShareWith.replace(/\s*/g, "").split(",")
```

Comment:
Just thinking about this interface, and how we report back to the user if say, they enter 1 correct email and 1 email that isn't registered with Naomi. 

Would it be better to have multiple inputs, one for each email, so that success/error can be reported per recipient? Probably right?

Reply:
Ok, will leave the input styling as is. In terms of implementation, I could change the endpoint to accept a list of emails, but then if some fail and some succeed, what status code should it report? Alternatively we do a separate POST for each email, showing concatenated success/error messages from each call. Thoughts?

Reply:
Hm, good questions. I guess it would still be a 400 because it failed overall because of invalid input, but since it's affected the state by adding the valid emails that seems confusing. Maybe multiple calls would be cleaner.

Reply:
I guess another option is have a single endpoint but if any of the emails are invalid, the whole request fails, i.e. nothing persisted to the db?